### ISSUE TITLE
Build cannot find package.json via SmooshPackageJSON (embroider:core:smoosh-package-json)

### ISSUE THREAD

**Issue by knownasilya** [embroider-build/embroider#643]
```
ember build --watch
Environment: development
Build Error (OneShot)

ENOENT: no such file or directory, open '/var/folders/g5/dvq_p0010s5b2ys8bwd7czdr0000gp/T/broccoli-90580jYsLRGENdlm6/out-14-broccoli_merge_trees/package.json'
        at SmooshPackageJSON (embroider:core:smoosh-package-json)
-~- created here: -~-
    at new Plugin (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/node_modules/broccoli-plugin/dist/index.js:45:33)
    at new SmooshPackageJSON (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/src/smoosh-package-json.js:12:9)
    at buildCompatAddon (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/src/build-compat-addon.js:43:24)
    at Object.cachedBuildCompatAddon [as default] (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/src/build-compat-addon.js:13:16)
    at movedAddons.map.oldPkg (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/src/compat-addons.js:45:105)
    at Array.map (<anonymous>)
    at CompatAddons.get tree [as tree] (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/compat/src/compat-addons.js:45:63)
    at CompatApp.augment (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/core/src/build-stage.js:57:64)
    at CompatApp.get tree (/Users/iradchenko/workspace/goodroute/client/node_modules/@embroider/core/src/build-stage.js:26:50)
    at CompatApp.<anonymous> (/Users/iradchenko/workspace/goodroute/client/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
-~- (end) -~-
```

Setup in app:

```js
const { Webpack } = require('@embroider/webpack');
  return require('@embroider/compat').compatBuild(app, Webpack, {
    staticAddonTestSupportTrees: true,
    staticAddonTrees: true,
    staticHelpers: true,
  });
```

ember-source@3.23.1
    "@embroider/compat": "^0.33.0",
    "@embroider/core": "^0.33.0",
    "@embroider/router": "^0.33.0",
    "@embroider/webpack": "^0.33.0",

**Comment by knownasilya** [comment]
Seems to happen when I update these packages to 0.7.0

```
    "@frontile/buttons": "^0.2.1",
    "@frontile/changeset-form": "^0.2.1",
    "@frontile/core": "^0.2.1",
    "@frontile/forms": "^0.2.1",
    "@frontile/notifications": "^0.2.1",
```

**Comment by mattmcmanus** [comment]
I can confirm that I'm also running into this with `0.35.0`

```
  "@embroider/compat": "^0.35.0",
  "@embroider/core": "^0.35.0",
  "@embroider/webpack": "^0.35.0",
```

**Comment by ef4** [comment]
Please patch `@embroider/compat` something like this so we can see which addon is responsible:


```diff
diff --git a/packages/compat/src/build-compat-addon.ts b/packages/compat/src/build-compat-addon.ts
index 72e1305..bb25952 100644
--- a/packages/compat/src/build-compat-addon.ts
+++ b/packages/compat/src/build-compat-addon.ts
@@ -42,7 +42,7 @@ function buildCompatAddon(originalPackage: Package, v1Cache: V1InstanceCache): N
   let needsSmooshing = oldPackages[0].hasAnyTrees();
   if (needsSmooshing) {
     let trees = oldPackages.map(pkg => pkg.v2Tree).reverse();
-    let smoosher = new SmooshPackageJSON(trees);
+    let smoosher = new SmooshPackageJSON(trees, { annotation: originalPackage.name });
     return broccoliMergeTrees([...trees, smoosher], { overwrite: true });
   } else {
     return oldPackages[0].v2Tree;
diff --git a/packages/compat/src/smoosh-package-json.ts b/packages/compat/src/smoosh-package-json.ts
index 2040fd2..9f40d19 100644
--- a/packages/compat/src/smoosh-package-json.ts
+++ b/packages/compat/src/smoosh-package-json.ts
@@ -5,9 +5,9 @@ import { join } from 'path';
 import { mergeWithUniq } from './merges';

 export default class SmooshPackageJSON extends Plugin {
-  constructor(inputTrees: Node[]) {
+  constructor(inputTrees: Node[], opts: { annotation?: string } = {}) {
     super(inputTrees, {
-      annotation: 'embroider:core:smoosh-package-json',
+      annotation: `embroider:core:smoosh-package-json:${opts?.annotation}`,
     });
   }
```

It's probably OK to just ignore the missing file instead of throwing, but it would be good to understand why one instance of an addon lacks a package.json.

**Comment by mattmcmanus** [comment]
My issue appeared to be caused by https://github.com/emberjs/ember-test-waiters and how it trys to expose itself to apps. Thanks to @ef4 [for the discovery](https://discord.com/channels/480462759797063690/568935504288940056/796825997735362610).

https://github.com/embroider-build/embroider/pull/663 should keep embroider from blowing up.

**Comment by mattmcmanus** [comment]
@knownasilya This should be fixed in `0.35.1`: https://github.com/embroider-build/embroider/blob/master/CHANGELOG.md#0351-2021-01-11
